presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1181,6 +1181,18 @@ private PhysicalOperation visitScanFilterAndProject(
                     channel++;
                 }
             }
+            //TODO: This is a simple hack, it will be replaced when we add ability to push down sampling into connectors.
+            // SYSTEM sampling is performed in the coordinator by dropping some random splits so process only for it.
+            else if (sourceNode instanceof SampleNode) {
+                SampleNode sampleNode = (SampleNode) sourceNode;
+                checkArgument(sampleNode.getSampleType() == SampleNode.Type.SYSTEM, "SYSTEM sampling type is only supported");
```

Comment:
I think it might be better to say something like:
```
checkArgument(sampleNode.getSampleType() == SampleNode.Type.SYSTEM, "%w sampling is not supported", sampleNode.getSampleType());

Reply:
Have updated it.